### ISSUE TITLE
Matplotlib regression in widget removal

### ISSUE THREAD

**Issue by jat255** [hyperspy/hyperspy#2965]
#### Describe the bug
Commit https://github.com/hyperspy/hyperspy/commit/02745991c338de35448be8ff98ff66c1e7577d51 modified widget removal code, and it appears to not work as intended. I had some code prior that would take the navigator image for a linescan plot, which would remove the navigation pointer. That's no longer working as of v1.7.

#### To Reproduce

```python
import hyperspy.api as hs
import numpy as np

sp = hs.datasets.example_signals.EDS_TEM_Spectrum()
s = hs.stack([sp * i for i in np.arange(.1, 1, .3)], new_axis_name='x')
s.plot()
s._plot.pointer.set_on(False)       # remove pointer
f = s._plot.navigator_plot.figure
f.savefig('/tmp/output.png')      # notice that pointer is in output (in 1.6.5, this works)
```

##### v1.7 output (with `matplotlib` v3.5.2)

![output](https://user-images.githubusercontent.com/1278301/174356887-5ced2437-af9a-46d1-9a93-358ad608841f.png)

##### v1.6.5 output (with `matplotlib` v3.4.3)

![output](https://user-images.githubusercontent.com/1278301/174357067-e1c68afb-3b03-4cb9-bb16-1f865566280c.png)

#### Python environment:
 - HyperSpy version: 1.7.0
 - Python version: 3.7.12
 - Matplotlib version: 3.5.2

#### Additional context

This appears to be because of the changes at https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/drawing/widget.py#L134-L138

`p.remove()` appears to not do anything by my investigation. Also, even though I'm on v3.5.2 of `matplotlib`, the `container.remove(p)` command still works. I presume this was added do to some sort of deprecation I cannot find...

**Comment by jat255** [comment]
I see this is related to this deprecation: https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.5.0.html#modification-of-axes-children-sublists

**Comment by ericpre** [comment]
Indeed, the way matplotlib remove artists has changed and if we didn't changed, we will have warning that this would break in matplotlib 3.6 or 3.7. However, this should work and the widget should be remove at the next draw.

What is used by matplotlib for the line widget can be summarised as:
```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots()

ax.plot([0, 1, 2])
# the horizontal widget is using axhline
l = ax.axhline(1.25, color='C1')

# remove the line
l.remove()
# trigger a draw to remove the line from the figure
fig.canvas.draw_idle()
```
This works as expected.

This does work as expected for a square pointer (navigator 2d):

```python
import hyperspy.api as hs
import numpy as np

s = hs.signals.Signal1D(np.arange(10*10*50).reshape(10, 10, 50))
s.plot()

# Remove pointer
s._plot.pointer.close()
# trigger a draw to remove the square widget from the figure
s._plot.navigator_plot.render_figure()
```

This does *not* (<-EDIT) work as expected for a line pointer (line 2d):

```python
import hyperspy.api as hs
import numpy as np

s = hs.signals.Signal1D(np.arange(10*50).reshape(10, 50))
s.plot()

# Remove pointer
s._plot.pointer.close()
# trigger a draw to remove the square widget from the figure
s._plot.navigator_plot.render_figure()
```

The main difference between the two last example is in the type of widget used, `HorizontalLineWidget` and `SquareWidget` and the type of matplotlib object used for the widget.

**Comment by jat255** [comment]
So the last example does not work on my system:

![image](https://user-images.githubusercontent.com/1278301/174839080-440e571f-4185-46da-bf40-bee240b1a91f.png)

![image](https://user-images.githubusercontent.com/1278301/174838880-c719b2fd-36b0-45f6-81aa-6ddfc85ee31e.png)

```
$ pip list | grep -E "hyperspy|matplotlib"
hyperspy                          1.7.0
matplotlib                        3.5.2
matplotlib-inline                 0.1.3
```

**Comment by ericpre** [comment]
Sorry, there was a typo in my [comment](https://github.com/hyperspy/hyperspy/issues/2965#issuecomment-1159457986) above, I haved edited it.

**Comment by jat255** [comment]
Ok, glad to know I'm not crazy! I'm curious why one type of marker works and the other doesn't. Maybe it's an underlying bug in matplotlib? I'd have to look into it more.

**Comment by ericpre** [comment]
Should be fixed in #3008.
